cocotb/runner.py

```diff
@@ -934,13 +959,201 @@ def clean(recursive: bool = False) -> None:
     dir = os.getcwd()
 
     def rm_clean() -> None:
-        build_dir = os.path.join(dir, "sim_build")
-        if os.path.isdir(build_dir):
-            print("INFO: Removing:", build_dir)
+        build_dir = Path(dir) / "sim_build"
+        if Path(build_dir).is_dir():
+            print("INFO: Removing", build_dir)
             shutil.rmtree(build_dir, ignore_errors=True)
 
     rm_clean()
 
     if recursive:
         for dir, _, _ in os.walk(dir):
             rm_clean()
+
+
+class ParseKwargs(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        setattr(namespace, self.dest, dict())
+        for value in values:
+            key, value = value.split("=")
+            getattr(namespace, self.dest)[key] = value
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter)
+
+    parser.add_argument(
+        "--sim", default="icarus", help="The simulator to use (default: %(default)s)"
+    )
+
+    parser.add_argument(
+        "--always", help="Always run the build step", action="store_true"
+    )
+    parser.add_argument(
+        "--build-dir",
+        type=Path,
+        default="sim_build",
+        help="Directory to run the build step in (default: %(default)s)",
+    )
+    parser.add_argument(
+        "--defines",
+        nargs="*",
+        metavar=("NAME=VALUE"),
+        action=ParseKwargs,
+        help="Defines to set",
+    )
+    parser.add_argument(
+        "--extra-args",
+        nargs="*",
+        default=[],
+        help="Extra arguments for the simulator",
+    )
+    parser.add_argument(
+        "--extra-env",
+        nargs="*",
+        default=[],
+        help="Extra environment variables to set",
+    )
+    parser.add_argument("--gui", help="Run with simulator GUI", action="store_true")
+    parser.add_argument(
+        "--includes",
+        nargs="*",
+        type=Path,
+        default=[],
+        help="Verilog include directories",
+    )
+    parser.add_argument(
+        "--library-name",
+        default="work",
+        help="Library name to compile into (default: %(default)s)",
+    )
+    parser.add_argument(
+        "--parameters",
+        "--generics",  # TODO: discuss this alt name
+        dest="parameters",
+        nargs="*",
+        metavar=("NAME=VALUE"),
+        action=ParseKwargs,
+        help="Verilog parameters or VHDL generics",
+    )
+    parser.add_argument(
+        "--plus-args",
+        "--plusargs",  # TODO: discuss this alt name
```

Comment:
Much prefer `--plusargs`.

Reply:
I agree, but lets move renaming to a different PR after this one is through.

Reply:
Done this anyway now.